internal/services/maintenance/maintenance_assignment_virtual_machine_scale_set_resource.go

```diff
@@ -41,18 +42,19 @@ func resourceArmMaintenanceAssignmentVirtualMachineScaleSet() *pluginsdk.Resourc
 			"location": azure.SchemaLocation(),
 
 			"maintenance_configuration_id": {
-				Type:         pluginsdk.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: validate.MaintenanceConfigurationID,
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateFunc:     maintenanceconfigurations.ValidateMaintenanceConfigurationID,
+				DiffSuppressFunc: suppress.CaseDifference, // TODO remove in 4.0
```

Comment:
how come we need to ignore the case here? and we should be wrapping it in a 4.0 beta block so it is removed when true